### PR TITLE
FLUID-6555: add initial configuration for CSS and Sass linting

### PR DESCRIPTION
**Pull request by greatislander**
This is an initial attempt at [FLUID-6555](https://issues.fluidproject.org/browse/FLUID-6555) which uses the Stylelint configuration from [inclusive-design/idrc](https://github.com/inclusive-design/idrc/blob/12712dd/stylelint.config.js). The configuration rules should be able to be overridden as documented in #17. Hoping to start the discussion from here, including an evaluation of these rules to determine if they are suitable as a starting point for Fluid Project… projects!

### CODE REVIEW

**Comment by the-t-in-rtf** [comment]
Based on our experience with Justin's recent pull, I think a good litmus test of this would be to create a "companion pull" against Infusion that uses the work in progress here for its linting.  That would allow us to confirm whether there are linting errors in Infusion itself and to talk about either adjusting rules or fixing errors.

**Comment by greatislander** [comment]
@the-t-in-rtf Yes, that makes sense. In [FLUID-6551](https://issues.fluidproject.org/projects/FLUID/issues/FLUID-6551) I'd proposed exploring alternatives to Stylus; this PR currently doesn't lint Stylus files (there's an experimental Stylus linting extension for Stylelint but I haven't tested it) so it may not have any impact on Infusion.

**Comment by the-t-in-rtf** [comment]
At least until the UI components are moved out of the repo, there are still raw CSS files that should be included in linting.  See [my comment on FLUID-6551](https://issues.fluidproject.org/browse/FLUID-6551?focusedCommentId=42007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-42007).

It would also be nice to have support for Stylus, in looking at your list on FLUID-6551, I realised how many projects would benefit from having that.  How much work are we talking?

**Comment by greatislander** [comment]
@the-t-in-rtf I'm going to try updating this PR with some Stylus code and the stylus extension and see if it works. I'll also open a PR against infusion once I've done that.

**Comment by greatislander** [comment]
@the-t-in-rtf So, I've tested a bit, and it's not going to work to have a shared config for CSS, SCSS and Stylus. We can have a shared CSS/SCSS configuration as the syntaxes are more compatible but we would need a separate configuration to lint Stylus.

**Comment by greatislander** [comment]
@the-t-in-rtf I'm going to explore the use of [stylint](https://github.com/SimenB/stylint) (not to be confused with [stylelint](https://stylelint.io)) for independently linting Stylus files. (Props to @blueslug for suggesting I look for other linters that could handle Stylus.)

**Comment by the-t-in-rtf** [comment]
Glad to hear it.   I'll wait to hear from you.

**Comment by greatislander** [comment]
@the-t-in-rtf After further investigating, I don't believe that stylint is a viable option. It hasn't been updated in over a year and has a number of reported issues; most significantly the `stylint-stylish` reporter seems not to work properly with `fluid-grunt-lint-all` and hasn't been updated since 2018.

There is an "official" linter: https://github.com/stylus/stlint

However, it doesn't have any existing Grunt integrations and if the ultimate goal is what you've outlined in #21, I'm not sure it makes sense to spend time developing a Grunt integration for stlint. Thoughts?

**Comment by the-t-in-rtf** [comment]
Thank you for putting in the time to research this, @greatislander.  Adding at least some CSS coverage seems better than continuing with none.  I will take a look at the other pull shortly.

File: .stylelintrc [hunk 1]
```diff
@@ -4,6 +4,8 @@
 		"stylelint-order"
 	],
 	rules: {
+		"block-opening-brace-newline-after": "always",
+        "block-closing-brace-newline-before": "always",
```
Comment (the-t-in-rtf) [inline .stylelintrc]: Seems like this file needs to be added to the JSON linting.

Comment (greatislander) [reply .stylelintrc]: Resolved in 7319e5b.

**Comment by the-t-in-rtf** [comment]
I think the [associated Infusion pull](https://github.com/fluid-project/infusion/pull/1014) is looking good, I'll check in with @amb26 and see if we can round out this pair of pulls soon.   If the work on the standard config is likely to be available, I think it would make sense to:

1.  Review and merge the central config.
2. Cut a release of the central config.
3. Update this pull and the infusion pull to use the central config.
4. Merge this pull and cut a release of fluid-grunt-lint-all
4. Update the Infusion pull to point to the release of fluid-grunt-lint-all.
5. Merge the Infusion pull.

**Comment by greatislander** [comment]
@the-t-in-rtf Steps 1-3 are now complete (see: [fluid-project/stylelint-config-fluid](https://github.com/fluid-project/stylelint-config-fluid)).

**Comment by the-t-in-rtf** [comment]
> Update this pull and the infusion pull to use the central config.

The infusion pull still uses a hard-coded `.stylelintrc` file.  It would be good to confirm whether you have to bring in the config as a dev dependency in Infusion, or whether it will bring it in via fluid-grunt-lint-all.  I'm assuming it will have to be a dev dependency, but in any case we should at least have a consistent `.stylelintrc.json` file in the Infusion pull to confirm there are no additional surprises.

**Comment by greatislander** [comment]
@the-t-in-rtf Thanks for the reminder— I've updated that PR. It's still referencing this branch for `fluid-grunt-lint-all`.

**Pull request merged**